src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -405,10 +405,20 @@ ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
                                        uct_ib_mlx5_mmio_mode_t mmio_mode)
 {
 #if HAVE_DEVX
-    uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context, 0);
+    uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context,
+                                                UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);
     if (uar->uar == NULL) {
-        ucs_error("mlx5dv_devx_alloc_uar() failed: %m");
-        return UCS_ERR_NO_MEMORY;
+        if (UCT_IB_MLX5_UAR_ALLOC_TYPE_WC == UCT_IB_MLX5_UAR_ALLOC_TYPE_NC) {
+            ucs_error("mlx5dv_devx_alloc_uar(WC) failed: %m");
```

Comment:
1. print a debug message if WC failed and we're doing fallback
2. print actual flag value, ucs_error("mlx5dv_devx_alloc_uar(WC, flags=0x%x) failed: %m", UCT_IB_MLX5_UAR_ALLOC_TYPE_WC);